### DISCUSSION TITLE
How to host on a website

### DISCUSSION THREAD

**Discussion by curiousdork** [AppFlowy-IO/AppFlowy#283] [Technical help wanted]
Has anyone successfully gotten AppFlowy hosted on a website? If so, any chance y'all have it documented?

**Comment by BrutalCoding** [comment]
# Disclaimer
I've just discovered appflowy a couple of minutes ago and noticed the frontend is built with Flutter, a framework that I'm very familiar with. Its quite straight forward to host Flutter on the web, however, it's using Rust as well for some of its backend so that throws me a little bit off the hook. Sorry if none of this info will result into AppFlowy running on web! I will give it a try myself within a month. 

I have not looked into the source code of AppFlowy, I just wanted to give my thoughts straight away.

# What I would do (waste of time, but fun to try)
## Build Flutter web, without (Rust) backend running.. 
From on top of my head, you might wanna look into the following links to build Flutter as a webapp and host it online. However, although AppFlowy will probably compile and be hosted online successfully, it most likely won't work since the backend (Rust) is not hosted together with the frontend (Flutter). You will have to figure out how to fix that. A solution would be to host the backend somewhere else and change the Flutter (frontend) code API calls to point to where you hosted the backend. This is because the frontend + backend is not running on the same server in this case.

- This article showcases creating a new flutter app and hosting it on Firebase Hosting (I recommended it, and its free)
https://developer.school/tutorials/hosting-a-flutter-web-project-on-firebase-hosting

- Official docs, however it does not specially target Flutter, still useful though because of many references to get more in depth:
https://firebase.google.com/docs/hosting

## Result
Now, again, assuming AppFlowy is built with plugins/dependencies that are web compatible (1), it compiles and is hosted online now. However its should be broken because its depending on the backend (Rust). In that case I will probably look into hosting the docker container to make things much easier. An alternative solution would be to host the (Rust) backend somewhere, and then change the Flutter code to point its API's to where you hosted the backend.

# What you should probably do
## Host docker container

Read this doc from AppFlowy that explains how to build a container:
https://appflowy.gitbook.io/docs/essential-documentation/install-appflowy/installation-methods/installing-with-docker

The goal is to host the docker container online. This link shows how to host a docker container using Google Cloud Run (free tier should be plenty enough):
https://cloud.google.com/run/docs/quickstarts/deploy-container

## Result
Again, I haven't tried this, but assuming you got an AppFlowy docker container and you used Google Cloud Run to host the container which gives you an URL, its probably working by now.

# Let me know
Hope that this helps out someone or at least points someone to the right direction to get it running as an online webapp!

# References:
1. I havent checked if AppFlowy is web compatible, check out the pubspec.yaml file, you'll see all used plugins, look them up on pub.dev to see whether they are web compatible or not

**Reply by heyakyra** [reply]
The docker run requires X access…but shouldn't the backend run independently and just be hosted over an arbitrary port?

**Comment by sprklinginfo** [comment]
Has anyone tried to host it on a website successfully? new to Appflowy and Flutter and looking for a notion alternative to collaborate for a small team, read the docs and didn't find if it supports collaborators...

**Reply by heyakyra** [reply]
https://github.com/AppFlowy-IO/AppFlowy/issues/89
https://github.com/AppFlowy-IO/AppFlowy/issues/714

**Reply by sprklinginfo** [reply]
Thanks! Hopefully, it will be delivered in July as planned.

**Comment by mickey3721** [comment]
yes! we need an appflowy to running in web page mode
